aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/Orchestrator.java

```diff
@@ -94,32 +97,88 @@ public Orchestrator(
         this.storageObserver = new StorageObserver(localStorageAdapter, mutationOutbox);
     }
 
+    /**
+     * Checks whether the orchestrator is {@link OrchestratorStatus#STARTED}.
+     * @return True if the orchestrator is started, false otherwise.
+     */
+    public boolean isStarted() {
+        return OrchestratorStatus.STARTED.equals(status.get());
+    }
+
     /**
      * Start performing sync operations between the local storage adapter
      * and the remote GraphQL endpoint.
      * @return A Completable operation to start the sync engine orchestrator
      */
     @NonNull
     public Completable start() {
-        return mutationOutbox.load()
-            .andThen(Completable.fromAction(() -> {
-                storageObserver.startObservingStorageChanges();
-                subscriptionProcessor.startSubscriptions();
-                syncProcessor.hydrate().blockingAwait();
-                mutationProcessor.startDrainingMutationOutbox();
-                subscriptionProcessor.startDrainingMutationBuffer();
-                LOG.info("Cloud synchronization is now fully active.");
-            }));
+        // Only start if it's stopped.
+        if (OrchestratorStatus.STOPPED.equals(status.get())) {
```

Comment:
I really like the use of finite state machines in the iOS codebase.. we should nick some of that.